packages/ember-application/lib/system/application.js

```diff
@@ -793,35 +978,198 @@ Application.reopenClass({
 if (isEnabled('ember-application-visit')) {
   Application.reopen({
     /**
-      Creates a new instance of the application and instructs it to route to the
-      specified initial URL. This method returns a promise that will be resolved
-      once rendering is complete. That promise is resolved with the instance.
-
-      ```js
-      App.visit('/users').then(function(instance) {
-        var view = instance.view;
-        view.appendTo('#qunit-test-fixtures');
-      });
-     ```
+      Boot a new instance of `Ember.ApplicationInstance` for the current
+      application and navigate it to the given `url`. Returns a `Promise` that
+      resolves with the instance when the initial routing and rendering is
+      complete, or rejects with any error that occured during the boot process.
 
-      @method visit
-      @private
-    */
-    visit(url) {
-      var instance = this.buildInstance();
-      this.runInstanceInitializers(instance);
-
-      var renderPromise = new Ember.RSVP.Promise(function(res, rej) {
-        instance.didCreateRootView = function(view) {
-          instance.view = view;
-          res(instance);
+      When `autoboot` is disabled, calling `visit` would first cause boot the
+      application. See the documentation on the `boot` method for details.
+
+      This method also takes a hash of boot-time configuration options for
+      customizing the instance's behavior. See the documentation on
+      `Ember.ApplicationInstance.BootOptions` for details.
+
+      `Ember.ApplicationInstance.BootOptions` is an interface class that exists
+      purely to document the available options; you do not need to construct it
+      manually. Simply pass a regular JavaScript object containing of the
+      desired options:
+
+      ```javascript
+      MyApp.visit("/", { location: "none", rootElement: "#container" });
+      ```
+
+      All options are optional.
```

Comment:
:stuck_out_tongue: 

This is an artifact of the way I documented them using the "pseudo" `BootOptions` class, I wanted to make clear you do not have to set every property that's present on that documentation page, but on second thoughts the `@default` should already take care of that (and if not, the fact that this example doesn't set other things should imply that).

So I'll rm